### PR TITLE
[WIP] Spinners design

### PR DESCRIPTION
**Pull request by ColdVolcano**
Pretty much nothing has been done, will update as I keep doing things.
For now, spinners are not centered by default.

Extends #269 (please merge first).

### CODE REVIEW

File: osu.Game.Modes.Osu/Objects/Spinner.cs [hunk 1]
```diff
@@ -5,5 +5,8 @@ namespace osu.Game.Modes.Osu.Objects
 {
     public class Spinner : OsuHitObject
     {
+        public double EndPoint;
```
Comment (peppy) [inline osu.Game.Modes.Osu/Objects/Spinner.cs]: don't like this variable name. we'll probably want to change the base implementation of EndTime or Duration to allow setting it directly.

File: osu.Game.Modes.Osu/Objects/Drawables/Pieces/SpinnerDisc.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        private void discScale()
+        {
+            disc.ScaleTo(1 + (MathHelper.Clamp(((totalAngleSpinned / 360) / rotationsNeeded), 0, 1)/10), 100);
+        }
+
+        private float GetAngledDifference()
+        {
+            lastAngle = actualAngle ?? GetMouseAngledPosition();
+            Delay(1);
```
Comment (peppy) [inline osu.Game.Modes.Osu/Objects/Drawables/Pieces/SpinnerDisc.cs]: what is this for..?

Comment (ColdVolcano) [inline osu.Game.Modes.Osu/Objects/Drawables/Pieces/SpinnerDisc.cs]: To get differences in the angle that the mouse is on right now and the angle was last millisecond. I changed this approach locally, but haven't commited yet.

**Comment by peppy** [comment]
See #269

**Pull request closed without merging**